huber/tests/test_driver.py

```diff
@@ -51,3 +53,24 @@ async def test_get_data(huber_driver, expected_data):
     with mock.patch('random.random', lambda: fixed_random), \
          mock.patch('random.choice', lambda arg: fixed_choice):
         assert expected_data == await huber_driver.get()
+
+
+@mock.patch('huber.Bath', Bath)
+def test_driver_cli(capsys, expected_data):
+    """Confirm the commandline interface works."""
+    with mock.patch('random.random', lambda: fixed_random), \
+         mock.patch('random.choice', lambda arg: fixed_choice):
+        command_line(['fakeip'])
+        captured = loads(capsys.readouterr().out)
+        assert expected_data == captured
```

Comment:
It's actually not terrible, since `pytest` gives a halfway-usable diff.

```
>           assert expected_data == captured
E           AssertionError: assert {'fill': 0.08...7964424}, ...} == {'fill': 0.08...7964424}, ...}
E             Omitting 5 identical items, use -vv to show
E             Differing items:
E             {'temperature': {'bath': 23.41, 'process': 22.71, 'setpoint': 1.23}} != {'temperature': {'bath': 23.49, 'process': 22.71, 'setpoint': 1.23}}
E             Use -v to get more diff

huber/tests/test_driver.py
```

Sorting is probably worthwhile, though.

Reply:
Right! I forgot this was Pytest that does extra magic for you (too used to Go at this point where this kinda thing isn't free).

Reply:
I tried sorting with the hilarious `loads(dumps(loads(capsys.readouterr().out), sort_keys=True))`

It was pointless, since the order doesn't affect the equality operator, and the dicts are already generated in the same order.